lib/redis/event_handler.js

```diff
@@ -39,6 +39,25 @@ exports.connectHandler = function (self) {
 
     self.initParser();
 
+    self.lastWriteTime = null;
+    if (typeof self.options.timeoutPerCommand === 'number') {
+      var timeoutPerCommand = self.options.timeoutPerCommand;
+      debug('Per-command timeout set to %s', self.options.timeoutPerCommand);
+      self.timeoutPerCommand = setInterval(() => {
+        var commandQueueLength = self.commandQueue.length;
+        if (
+          commandQueueLength > 0 &&
+          Date.now() - self.lastWriteTime > self.options.timeoutPerCommand
+        ) {
+          debug('Command timed out. Pending commands: %s', commandQueueLength);
+          var err = new Error('Command timed out');
+          self.flushQueue(err, { offlineQueue: false });
+          self.silentEmit('error', err);
```

Comment:
Is emitting an `error` event sufficient?  In my usage, I'm always supplying a callback in the form of:
```
redis.get(key, function (err, reply) {
    if (err) {
        // handle error
    } else {
        // handle data
    }
});
```
While I do also handle the `error` event, that handling is somewhat generic.  It would be better if the callback was called with the error so that specific error handling could be invoked.